### PR TITLE
[aptos-cli] Add testing for the Move CLI commands and cleanup e2e testing

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
This adds tests for the Move CLI to ensure it doesn't break.  It also cleans up the way the package system is initialized.

### Test Plan
Well, I'm adding tests, so those.

Also a manual test:
```
$ aptos move init --package-dir testing --name myTest --named-addresses Name=_
"testing/Move.toml" already exists, are you sure you want to overwrite it? [yes/no] >
yes
{
  "Result": "Success"
}

$ cat testing/Move.toml 
[package]
name = 'myTest'
version = '0.0.0'

[addresses]
Name = '_'
[dependencies.AptosFramework]
git = 'https://github.com/aptos-labs/aptos-core.git'
rev = 'devnet'
subdir = 'aptos-move/framework/aptos-framework'
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2647)

### CODE REVIEW

File: crates/aptos/src/common/types.rs [hunk 1]
```diff
@@ -747,6 +747,45 @@ pub fn load_account_arg(str: &str) -> Result<AccountAddress, CliError> {
...
+        AccountAddress::from_hex_literal(str)
+            .map(Some)
+            .map_err(|err| {
+                CliError::CommandArgumentError(format!("Failed to parse AccountAddress {}", err))
+            })
+    } else if let Ok(account_address) = AccountAddress::from_str(str) {
+        Ok(Some(account_address))
+    } else if let Some(Some(private_key)) = CliConfig::load_profile(str)?.map(|p| p.private_key) {
+        let public_key = private_key.public_key();
+        Ok(Some(account_address_from_public_key(&public_key)))
```
Comment (davidiw) [inline crates/aptos/src/common/types.rs:780]: why add this when we already have this: https://github.com/aptos-labs/aptos-core/blob/21a0c247634c7afbd6acf41e0d557e92943a1c75/types/src/transaction/authenticator.rs#L399

I guess you have that, but might it make more sense to improve the auth key api instead?

File: crates/aptos/src/test/mod.rs [hunk 2]
```diff
@@ -46,10 +47,28 @@ use serde_json::Value;
 use std::collections::BTreeMap;
 use std::path::PathBuf;
 use std::{str::FromStr, time::Duration};
+use thiserror::private::PathAsDisplay;
 use tokio::time::{sleep, Instant};
 
 pub const INVALID_ACCOUNT: &str = "0xDEADBEEFCAFEBABE";
 
+pub const FIRST_MOVE_FILE: &str = "
```
Comment (davidiw) [inline crates/aptos/src/test/mod.rs:55]: maybe include_str?

Comment (davidiw) [reply crates/aptos/src/test/mod.rs:55]: also this isn't used in this PR

**Review by davidiw — APPROVED** [review]

File: crates/aptos/src/common/types.rs (hunk 1 above)
Comment (gregnazario) [reply crates/aptos/src/common/types.rs:780]: Can fix this later, it's been like this for a while

File: crates/aptos/src/test/mod.rs (hunk 2 above)
Comment (gregnazario) [reply crates/aptos/src/test/mod.rs:55]: Yeah, it's no longer used but I believe I removed it unless you're commenting on a specific commit

**Pull request merged**